src/edu/washington/escience/myria/operator/agg/Aggregate.java

```diff
@@ -102,6 +105,16 @@ protected TupleBatch fetchNextReady() throws DbException {
       tb = child.nextReady();
     }
     if (child.eos()) {
+      if (!COUNTALL_ON_EMPTY
```

Comment:
Would it be more broadly useful to have a boolean getter on `Operator` indicating whether an operator had ever returned any results (i.e. `nextReady() != null`)? If that could be reused elsewhere, it would be better than an ad-hoc flag.